app/rocket.py

```diff
@@ -40,63 +41,62 @@ def __init__(self):
 
     def sleep_state(self):
         if time.time() % 100 == 0:
-            comms.wake()
+            self.device_factory.comms.wake()
 
-            if comms.read('wake'):
+            if self.device_factory.comms.read('wake'):
                 return self.wake()
 
-            comms.sleep()
+            self.device_factory.comms.sleep()
 
 
     def ground_state(self):
         if self.kin_init == True:
+            self.kinetics.activate()
             self.kinetics.run()
             self.kin_init = False
 
         #TODO: init test systems
-        if comms.read('sleep'):
+        if self.device_factory.comms.read('sleep'):
             return self.sleep()
 
-        if self.kinetics.acceleration['z'] > 0:
+        if self.kinetics.acceleration()['z'] > 0:
             return self.launch()
 
 
     def powered_state(self):
         if self.timer_init == True:
             self.starttime = time.time()
             self.timer_init = False
-        #TODO: read IMU  
-        #TODO: z_accel <-9.81 OR t > 3s
+ 
         if (self.kinetics.acceleration()['z'] < -9.81) or ((time.time() - self.starttime) > 3):
```

Comment:
Why 3 seconds?

Reply:
incase the accelerometer dies, we can still track state based on the simulation data